modules/schema/src/main/java/org/apache/ignite/internal/schema/definition/builder/ColumnDefinitionBuilderImpl.java

```diff
@@ -78,6 +79,9 @@ public ColumnDefinitionBuilderImpl withHints(Map<String, String> hints) {
     /** {@inheritDoc} */
     @Override
     public ColumnDefinition build() {
-        return new ColumnDefinitionImpl(colName, colType, nullable, defValExpr);
+        var defaultSupp = defValExpr == null
```

Comment:
should we assert that nullable false for defValExpr = null?
May be need distinguish explicit and implicit null for the case

Reply:
> should we assert that nullable false for defValExpr = null?

I don't think so. `nullable==false && defValExpr==null` means that the user MUST provide value for every such column in insert tuple.

> May be need distinguish explicit and implicit null for the case

Could you please elaborate on why do you think this is important to distinguish between implicit and explicit null?

Reply:
I would say the following situation:
ColumnDefinitionBuilderImpl builder = ......;
builder.asNullable(false);
builder.withDefaultValueExpression(null);

What does it mean? For me it user mistake and we should say about it. But maybe you are right and we just should interpret it as the column value should be define during insert.

Reply:
This `create table t (c1 int, c2 int not null default null)` is a valid CREATE TABLE statement. Your example is the very same, but via Java API